### PR TITLE
Added HW_12 homework 1 and homework 2 files

### PR DESCRIPTION
**Pull request by Sapphireexe**
Finished homework 12:
part 1,
part 2.
All task are ready to review.

### CODE REVIEW

File: HW_12/hw_12_2.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  ["4", 5], // повторяющийся ключ, значит это значение должно быть в результирующем объекте
+]);
+
+console.log(result); //{ '1': 1, '2': 2, '3': 3, '4': 5 }
+*/
+
+function generateObject<T> (ArrayOfPairs: [string, T][]): {[key: string]: T} {
+  const resultObject: {[key: string]: T} = {};
+  for (const pair of ArrayOfPairs) {
+    resultObject[pair[0]] = pair[1];
```
Comment (VladimirLQA) [inline HW_12/hw_12_2.ts:51]:
```suggestion
  for (const [key, value] of arrayOfPairs) {
    resultObject[key] = value;
```

File: HW_12/hw_12_2.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  ["2", 2],
+  ["3", 3],
+  ["4", 4],
+  ["4", 5], // повторяющийся ключ, значит это значение должно быть в результирующем объекте
+]);
+
+console.log(result); //{ '1': 1, '2': 2, '3': 3, '4': 5 }
+*/
+
+function generateObject<T> (ArrayOfPairs: [string, T][]): {[key: string]: T} {
```
Comment (VladimirLQA) [inline HW_12/hw_12_2.ts:48]:
```suggestion
function generateObject<T> (arrayOfPairs: [string, T][]): {[key: string]: T} {
```

File: HW_12/hw_12_2.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  const resultObject: {[key: string]: T} = {};
+  for (const pair of ArrayOfPairs) {
+    resultObject[pair[0]] = pair[1];
+  }
+  return resultObject;
```
Comment (VladimirLQA) [inline HW_12/hw_12_2.ts:53]: Alternative way is just to use `return Object.fromEntries(arrayOfPairs)`

**Review by VladimirLQA — APPROVED** [review]

File: HW_12/hw_12_2.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+   Затипизировать надо саму функцию и коллбэк.
+   Создать реализацию функции map, принимающую массив чисел 1-5, возвращающую новый массив, 
+   где каждый каждый элемент - это элемент исходного массива умноженный на его индекс
+   Пример:
+   map([1,2,3,4,5], callback) => [0,2,6,12,20]
+*/
+
+type Callback<T, U> = (element: T, index: number, array: T[]) => U;
+
+function map<T, U> (array: T[], callback: Callback<T, U>): U[] {
```
Comment (anatoly-karpovich) [inline HW_12/hw_12_2.ts:13]: I hope generics are clear for now)

**Review by anatoly-karpovich — APPROVED** [review]

**Comment by anatoly-karpovich** [comment]
Great job!

**Pull request merged**